### PR TITLE
[Migration] migrate dev kit samples

### PR DESCRIPTION
**Pull request by hermanwenhe**
|        Q        |                    A                    |
| --------------- | --------------------------------------- |
| Bug fix?        | No                               |
| New feature?    | No                               |
| New sample?     | Yes                             |
| Related issues? |   |

## What's in this Pull Request?

Migrate the dev kit samples in https://github.com/OfficeDev/Office-Samples to this repo.

## Guidance

* Add `.config/samples-config-v1.json`, this config json file is for Dev Kit to parse all available samples under Samples folder.
* Add samples into Samples folder
* Add `CONTRIBUTING.md` for adding new samples to the sample gallery of Dev Kit.

### CODE REVIEW

File: .config/samples-config-v1.json [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            "thumbnailPath": "assets/thumbnail.png",
+            "gifPath": "assets/sampleDemo.gif",
+            "suggested": false
+        },
+        {
+            "id": "Word-Add-in-WritingAssistant",
+            "onboardDate": "2024-03-06",
+            "title": "Word Writing Assistant",
+            "shortId": "Word Writing Assistant",
+            "shortDescription": "A Word Add-in to assist writing with word document.",
```
Comment (AlexJerabek) [inline .config/samples-config-v1.json:81]: Some branding notes:

- It's "Word add-in", not "Word Add-in". The brand term is "Office Add-ins", so in this case we're talking about an add-in for Word.
- Always capitalize "Word" when referring to the application.
```suggestion
            "shortDescription": "A Word add-in to assist with writing a Word document.",
```

File: .config/samples-config-v1.json [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            "fullDescription": "This add-in demonstrates Word add-in capabilities to help users check errors, improve writing, and rephrase content.",
```
Comment (AlexJerabek) [inline .config/samples-config-v1.json:82]: Is this sample showing how the inherent capabilities of Word add-ins (i.e., Office.js) can help? Or that Word add-ins let you connect to third-party libraries that do this?

File: .config/samples-config-v1.json [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            "types": [
+                "Word"
+            ],
+            "tags": [
+                "TS",
+                "Writing Assistant",
```
Comment (AlexJerabek) [inline .config/samples-config-v1.json:88]: I'm not sure what the value of tags that repeat the title are? I don't think we're likely to have multiple samples tagged with "Writing Assistant", are we?

File: .config/samples-config-v1.json [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+            "thumbnailPath": "assets/thumbnail.png",
+            "gifPath": "assets/sampleDemo.gif",
+            "suggested": false
+        },
+        {
+            "id": "Excel-Add-in-ChartAPI-Analysis-Data",
+            "onboardDate": "2024-02-27",
+            "title": "Use the Excel Chart APIs to analyze data",
+            "shortId": "Chart API analyse data",
+            "shortDescription": "Using chart API to analyse data and showout using chart.",
```
Comment (AlexJerabek) [inline .config/samples-config-v1.json:100]: I'm not sure what "showout" means.

File: .config/samples-config-v1.json [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+            "fullDescription": "The sample add-in demonstrates Excel add-in capablities to help users using chart API to analyse data.",
```
Comment (AlexJerabek) [inline .config/samples-config-v1.json:101]: Here and elsewhere, the en-US spelling is "analyze". We should standardize to that.

File: Samples/Excel-Add-in-ChartAPI-Analysis-Data/RUN_WITH_EXTENSION.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
+# Run the add-in with Office Add-ins Development Kit extension
+We recommend you try this sample by using the [Office Add-ins Development Kit](https://marketplace.visualstudio.com/items?itemName=msoffice.microsoft-office-add-in-debugger). The Office Add-ins Development Kit is an end-to-end developer tool for building Office Add-ins. It helps create, run, and debug an Office Add-in.
```
Comment (AlexJerabek) [inline Samples/Excel-Add-in-ChartAPI-Analysis-Data/RUN_WITH_EXTENSION.md:2]: Nit: add blank lines after headers for better readability when editing.
```suggestion
# Run the add-in with Office Add-ins Development Kit extension

We recommend you try this sample by using the [Office Add-ins Development Kit](https://marketplace.visualstudio.com/items?itemName=msoffice.microsoft-office-add-in-debugger). The Office Add-ins Development Kit is an end-to-end developer tool for building Office Add-ins. It helps create, run, and debug an Office Add-in.
```

File: Samples/Excel-Add-in-ChartAPI-Analysis-Data/RUN_WITH_EXTENSION.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    <img src="./assets/devkit_preview.png" width="500" alt="Screenshot shows Preview your Office add-in in Office Add-ins Development Kit"/>
+
+    The extension then checks that the prerequisites are met before debugging starts. Check the terminal for detailed information if there are issues with your environment. After this process, the Excel desktop application launches and opens a new workbook with the sample add-in.
+
+1. **Stop Previewing Your Office Add-in**
+
+    Once you are finished testing and debugging the add-in, select **Stop Previewing Your Office Add-in**. This closes the web server and removes the add-in from the registry and cache.
+
+## Explore the add-in code
+To explore an Office add-in project, you can start with the key files listed below.
```
Comment (AlexJerabek) [inline Samples/Excel-Add-in-ChartAPI-Analysis-Data/RUN_WITH_EXTENSION.md:21]:
```suggestion
## Explore the add-in code

To explore an Office add-in project, you can start with the key files listed below.
```

File: Samples/Excel-Add-in-ChartAPI-Analysis-Data/RUN_WITH_EXTENSION.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+- Close the previous web server started for the sample with the **Stop Previewing Your Office Add-in** Office Add-ins Development Kit extension option.
+
+If you still have problems, see [troubleshoot development errors](https://learn.microsoft.com//office/dev/add-ins/testing/troubleshoot-development-errors) or [create a GitHub issue](https://aka.ms/officedevkitnewissue) and we'll help you.  
+
+For information on running the sample on Excel on the web, see [Sideload Office Add-ins to Office on the web](https://learn.microsoft.com/office/dev/add-ins/testing/sideload-office-add-ins-for-testing).
+
+For information on debugging on older versions of Office, see [Debug add-ins using developer tools in Microsoft Edge Legacy](https://learn.microsoft.com/office/dev/add-ins/testing/debug-add-ins-using-devtools-edge-legacy).
+
+## See also
+Once you understand the sample, make it your own! All the information about Office Add-ins is found in our [official documentation](https://learn.microsoft.com/office/dev/add-ins/overview/office-add-ins). You can also explore more samples in the Office Add-ins Development Kit. Select **View Samples** to see more samples of real-world scenarios.
```
Comment (AlexJerabek) [inline Samples/Excel-Add-in-ChartAPI-Analysis-Data/RUN_WITH_EXTENSION.md:42]:
```suggestion
## See also

Once you understand the sample, make it your own! All the information about Office Add-ins is found in our [official documentation](https://learn.microsoft.com/office/dev/add-ins/overview/office-add-ins). You can also explore more samples in the Office Add-ins Development Kit. Select **View Samples** to see more samples of real-world scenarios.
```

File: Samples/Excel-Add-in-ChartAPI-Analysis-Data/manifest.xml [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+        <bt:Url id="Commands.Url" DefaultValue="https://localhost:3000/commands.html"/>
+        <bt:Url id="Taskpane.Url" DefaultValue="https://localhost:3000/taskpane.html"/>
+      </bt:Urls>
+      <bt:ShortStrings>
+        <bt:String id="GetStarted.Title" DefaultValue="Get started with your sample add-in!"/>
+        <bt:String id="CommandsGroup.Label" DefaultValue="Commands Group"/>
+        <bt:String id="TaskpaneButton.Label" DefaultValue="Show Taskpane"/>
+      </bt:ShortStrings>
+      <bt:LongStrings>
+        <bt:String id="GetStarted.Description" DefaultValue="Your sample add-in loaded succesfully. Go to the HOME tab and click the 'Show Taskpane' button to get started."/>
```
Comment (AlexJerabek) [inline Samples/Excel-Add-in-ChartAPI-Analysis-Data/manifest.xml:80]: I thought we fixed this in the other repo. Please scrub this PR for this same typo.
```suggestion
        <bt:String id="GetStarted.Description" DefaultValue="Your sample add-in loaded successfully. Go to the HOME tab and click the 'Show Taskpane' button to get started."/>
```

File: Samples/Excel-Add-in-ChartAPI-Analysis-Data/src/commands/commands.html [hunk 10]
```diff
@@ -0,0 +1,18 @@
+<!-- Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT License. -->
```
Comment (AlexJerabek) [inline Samples/Excel-Add-in-ChartAPI-Analysis-Data/src/commands/commands.html:1]: Please remove command files from projects that don't need them.

File: .config/CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,21 @@
+# Contributing
+## Branch
+Default branch is `dev`. You should set `dev` as the target branch of PRs. `dev` will be merged into `main` regularly after testing.
+
+`main` branch is the release branch, which means all content in this branch will be directly consumed by the [Office Add-ins Development Kit](https://marketplace.visualstudio.com/items?itemName=msoffice.microsoft-office-add-in-debugger)'s sample gallery:
+![alt text](assets/sample_gallery.png)
+
+## Testing
+TBD
```
[the remark this answers is not in the document]
Comment (hermanwenhe) [reply .config/CONTRIBUTING.md:9]: I need this PR merged and also merged into main to build a Dev Kit package and check it in for testing. Will check in this section once migration is done.

File: Samples/Excel-Add-in-ChartAPI-Analysis-Data/README.md [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+<img src="./assets/thumbnail.png" width="800" alt="A workbook with tables and charts generated by the add-in.">
+
+This sample add-in demonstrates Excel add-in capabilities to help users using chart API to analyze data.
+
+## Features
+
+- Use Table and Range APIs to insert sample data.
+- Use Chart APIs to help with data analysis, such as chart formatting, trend lines, and top-value calculations.
+
+## How to run this sample
```
[the remark this answers is not in the document]
Comment (hermanwenhe) [reply Samples/Excel-Add-in-ChartAPI-Analysis-Data/README.md:12]: It's technically feasible, but the link should be added after these samples are already in the main branch.

File: .config/CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+
+## Add new sample
+1. Add a new project folder under Samples directory.
+    1. Make sure it's runnable and well tested.
```
[the remark this answers is not in the document]
Comment (hermanwenhe) [reply .config/CONTRIBUTING.md:13]: I totally agree that we should have a well-organized standard for testing and checking in new samples into the sample gallery. I don't have much opinion on this as engineer perspective. I would recommend leveraging the current sample checkin guideline of this repo. Maybe @davidchesnut can help on this?

File: .config/samples-config-v1.json (hunk 3 above)
Comment (hermanwenhe) [reply .config/samples-config-v1.json:88]: I don't the specific plan of adding new samples... But I think you guys can review and refine these tags or other properties in the config in a more thoughtful way once the migration is done.

File: .config/samples-config-v1.json (hunk 2 above)
Comment (hermanwenhe) [reply .config/samples-config-v1.json:82]: I would guess the writing assistant is one of the major scenarios of Word JS APIs so that a showcase of how to use these APIs might help. I think this question should better be answered by our PM @MingjiaLiu1995 .

File: .config/CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,21 @@
+# Contributing
+## Branch
+Default branch is `dev`. You should set `dev` as the target branch of PRs. `dev` will be merged into `main` regularly after testing.
```
[the remark this answers is not in the document]
Comment (hermanwenhe) [reply .config/CONTRIBUTING.md:3]: Yeah, we should wait David to confirm. I want to clarify that the branching policy (`main` for release, `dev` for developing and testing) is required for the Dev Kit development. If we want to discuss other branching policy, we should also consider the feasibility and effort of code implementation of Dev Kit.

File: .config/samples-config-v1.json (hunk 3 above)
Comment (hermanwenhe) [reply .config/samples-config-v1.json:88]: Btw, this tags are useful for matching purpose when users try to search specific sample.

File: .config/samples-config-v1.json (hunk 4 above)
Comment (hermanwenhe) [reply .config/samples-config-v1.json]: I use "present" to replace this.

File: .config/samples-config-v1.json (hunk 5 above)
Comment (hermanwenhe) [reply .config/samples-config-v1.json]: Done

File: Samples/Excel-Add-in-ChartAPI-Analysis-Data/src/commands/commands.html (hunk 10 above)
Comment (hermanwenhe) [reply Samples/Excel-Add-in-ChartAPI-Analysis-Data/src/commands/commands.html]: Done

File: .config/CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+1. Add a new project folder under Samples directory.
+    1. Make sure it's runnable and well tested.
+    2. Add README.md and RUN_WITH_EXTENSION.md. You can use the README_TEMPLATE.md under root directory.
+2. Add a new config to the [config file](samples-config-v1.json) with following format.
+    1. Fill in these values.
+    2. Make sure **id** has identical value with the folder name you just created
+    ![alt text](assets/config_format.png)
+
+### Config definition
+![alt text](assets/config_definition.png)
```
[the remark this answers is not in the document]
Comment (hermanwenhe) [reply .config/CONTRIBUTING.md:21]: Thanks for considering the accessibility. This guideline is more like a draft version for showing how it's done. I definitely agree that you should refine this to a more reliable and inclusive version.

File: .config/samples-config-v1.json (hunk 2 above)
Comment (MingjiaLiu1995) [reply .config/samples-config-v1.json:82]: I remember this sample is to showcase the Annotation APIs and the canvas card, including add annotation, ignore annotation, and show canvas card etc. Also add @wangyun-microsoft for notice.

File: .config/samples-config-v1.json (hunk 3 above)
Comment (AlexJerabek) [reply .config/samples-config-v1.json:88]: I would argue that it's the job of the product team to define how these tags are intended to be used, since they'll only be in the dev kit tool.

File: .config/CONTRIBUTING.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (AlexJerabek) [reply .config/CONTRIBUTING.md:3]: Pinging @davidchesnut since he's back from vacation.

File: .config/CONTRIBUTING.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (AlexJerabek) [reply .config/CONTRIBUTING.md:13]: This hints at the bigger issue with this PR: it's not really merging the dev kit samples with the existing samples. It's just adding them in the same place. If we're combining the two efforts, what we'd need from you are the instructions on how to add samples such that they get picked up by the tool.

File: .config/samples-config-v1.json (hunk 2 above)
Comment (AlexJerabek) [reply .config/samples-config-v1.json:82]: This sounds like a collection of snippets that is showing off APIs, rather than a customer-scenario. I would vote to not include it. We have Script Lab to show off individual APIs.

**Comment by AlexJerabek** [comment]
@hermanwenhe @MingjiaLiu1995, I responded to some of your comments. There are also some unaddressed issues I raised in previous iterations.

File: .config/samples-config-v1.json (hunk 2 above)
Comment (hermanwenhe) [reply .config/samples-config-v1.json]: @MingjiaLiu1995 @AlexJerabek  I removed Writing assistant for now. I'll add it again once you make an agreement.

File: .config/CONTRIBUTING.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (hermanwenhe) [reply .config/CONTRIBUTING.md:49]: I've added some part about how the config works and how to check in existing sample in this repo.

File: .config/samples-config-v1.json (hunk 3 above)
Comment (MingjiaLiu1995) [reply .config/samples-config-v1.json]: I agree removing "writing assistant" as it's repeated with title.
For tags, we should align with the filterOptions to keep it as simple, unless we get new requirements. So we have 3 types: Platform(WXPO), Language(TS/JS), Technology - Technology could be tricky, but I think we can start with common topics among developers such as CF, SSO, NAA etc.
I am also thinking about a tag of sample skill level, like beginner/intermediate/advanced, but maybe we postpone that conversation till we have more candidates.

File: .config/samples-config-v1.json (hunk 2 above)
Comment (AlexJerabek) [reply .config/samples-config-v1.json]: First off, if your document is too short, the "Step 2" button doesn't do anything. Without some guidance, users might assume the sample is broken when nothing happens.

Second, my original point about the description is that it makes it sound like Word APIs help with writing assistance. However, it seems more than we provide a couple APIs (annotations) that help writing assistants implement their add-ins. I don't find it very clear in the sample code where I'd replace parts with my own assistance model. Seems like too much is hard-coded at the moment.

Third, I find the use of arbitrary selections and text is confusing. I would be less confused with a sample piece of text with clear issues that could be analyzed. Like, if the sample searched for the word "very" and suggested that word be removed. Or some common programming acronym was replaced with the spell-out version.

File: .config/samples-config-v1.json [hunk 16]
```diff
@@ -0,0 +1,175 @@
+{
+    "filterOptions": {
+        "capabilities": [
+            "Word",
+            "Excel",
+            "PowerPoint"
+        ],
+        "languages": [
+            "TS",
+            "JS"
```
[the remark this answers is not in the document]
Comment (AlexJerabek) [reply .config/samples-config-v1.json:10]: @hermanwenhe, looks like a few of my comments haven't been looked at. I think are the ones under the "hidden" section on the main PR view.

File: .config/CONTRIBUTING.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (AlexJerabek) [reply .config/CONTRIBUTING.md:39]: @hermanwenhe, the `dev` branch works for us. However, can we split this into one PR per sample? It's hard to review the entire thing (GitHub lags).

[the remark this answers is not in the document]
Comment (davidchesnut) [reply .config/CONTRIBUTING.md:39]: @hermanwenhe is the dev branch for preview purposes in the devkit? If so that totally makes sense to have a dev branch. Or is there some other dependency for having the dev branch? Thanks!

[the remark this answers is not in the document]
Comment (hermanwenhe) [reply .config/CONTRIBUTING.md:39]: @davidchesnut dev branch is mainly for release purpose. Like samples in Teams Toolkit, they add the samples to the dev branch and do all the testing, then merge dev branch to main branch to release it to TTK developers once a while (usually a month).

@AlexJerabek OK, I can do the split.

File: .config/samples-config-v1.json (hunk 16 above)
[the remark this answers is not in the document]
Comment (hermanwenhe) [reply .config/samples-config-v1.json:10]: Yeah, I did miss it. I'll address these in the split PRs.
